Patch from Ubuntu's packaging 0.3.94-0ubuntu2:

* src/datetime-prefs-locations.c, src/datetime-prefs.c
  - Switch the parsing of the lat and long to be non-locale based as it's
    returned from the webservice in C, we want to parse it using the decimal
    instead of a comma.  This fixes incorrect timezone selection for users
    with a locale that uses a comma as a decimal separator  (LP: #773987)

--- indicator-datetime-0.3.94.orig/src/datetime-prefs.c
+++ indicator-datetime-0.3.94/src/datetime-prefs.c
@@ -613,11 +613,11 @@
                         -1);
 
     if (strlon != NULL && strlon[0] != 0) {
-      lon = strtod(strlon, NULL);
+      lon = g_ascii_strtod(strlon, NULL);
     }
 
     if (strlat != NULL && strlat[0] != 0) {
-      lat = strtod(strlat, NULL);
+      lat = g_ascii_strtod(strlat, NULL);
     }
 
     zone = cc_timezone_map_get_timezone_at_coords (self->priv->tzmap, lon, lat);
--- indicator-datetime-0.3.94.orig/src/datetime-prefs-locations.c
+++ indicator-datetime-0.3.94/src/datetime-prefs-locations.c
@@ -201,8 +201,8 @@
                         CC_TIMEZONE_COMPLETION_LONGITUDE, &strlon,
                         -1);
 
-    if (strlat && *strlat) lat = atof(strlat);
-    if (strlon && *strlon) lon = atof(strlon);
+    if (strlat && *strlat) lat = g_ascii_strtod(strlat, NULL);
+    if (strlon && *strlon) lon = g_ascii_strtod(strlon, NULL);
 
     CcTimezoneMap * tzmap = CC_TIMEZONE_MAP (g_object_get_data (G_OBJECT (widget), "tzmap"));
     g_free (zone);
